packages/smooth_app/lib/pages/input/agnostic_suggestion_manager.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/pages/input/brand_suggestion_manager.dart';
+
+// TODO(monsieurtanuki): there's probably a more elegant way to do it.
+/// Suggestion manager for "old" taxonomies and elastic search taxonomies.
+class AgnosticSuggestionManager {
+  AgnosticSuggestionManager.tagType(this.tagTypeSuggestionManager)
```

Comment:
Wouldn't a factory be a nicer API.
It would allow to hide the complexity of the implementation

Reply:
@g123k I honestly cannot see how a `factory` would help, but I would gladly see what you would write instead.
 Btw I wouldn't call the code of this file "complex" (one class, 28 lines of code, 2 fields, 1 method).